docs/api-reference/tasks-api.md

```diff
@@ -25,77 +25,1448 @@ sidebar_label: Tasks
+            "VARCHAR",
+            "VARCHAR",
+            "VARCHAR",
+            "VARCHAR",
+            "VARCHAR",
+            "VARCHAR"
+        ],
+        "context": {
+            "forceTimeChunkLock": true,
+            "useLineageBasedSegmentAllocation": true
+        },
+        "groupId": "query-32663269-ead9-405a-8eb6-0817a952ef47",
+        "dataSource": "wikipedia_api",
+        "resource": {
+            "availabilityGroup": "query-32663269-ead9-405a-8eb6-0817a952ef47",
+            "requiredCapacity": 1
+        }
+    }
+  }
+  ```
+
+</details>
+
+### Get task status
+
+#### URL
+<code class="getAPI">GET</code> `/druid/indexer/v1/task/{taskId}/status`
+
+Retrieves the status of a task given the task id. It returns a JSON object with the task's current status (`running`, `complete`, `waiting`, `pending`), task type, datasource, and other relevant metadata.
+
+#### Query parameters
+
+No query parameters.
+
+#### Responses 
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--200 SUCCESS-->
+<br/>
+*Successfully retrieved task status* 
+<!--404 NOT FOUND-->
+<br/>
+*Cannot find task with id* 
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample request
+
+The following examples shows how to retrieve the status of a task with the specified id `query-223549f8-b993-4483-b028-1b0d54713cad`.
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--cURL-->
+```shell
+curl "{domain}/druid/indexer/v1/task/query-223549f8-b993-4483-b028-1b0d54713cad/status"
+```
+<!--HTTP-->
+```HTTP
+GET /druid/indexer/v1/task/query-223549f8-b993-4483-b028-1b0d54713cad/status HTTP/1.1
+Host: {domain}
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+
+#### Sample response
+<details>
+  <summary>Click to show sample response</summary>
+
+  ```json
+  {
+    'task': 'query-223549f8-b993-4483-b028-1b0d54713cad',
+    'status': {
+      'id': 'query-223549f8-b993-4483-b028-1b0d54713cad',
+      'groupId': 'query-223549f8-b993-4483-b028-1b0d54713cad',
+      'type': 'query_controller',
+      'createdTime': '2023-06-22T22:11:28.367Z',
+      'queueInsertionTime': '1970-01-01T00:00:00.000Z',
+      'statusCode': 'RUNNING',
+      'status': 'RUNNING',
+      'runnerStatusCode': 'RUNNING',
+      'duration': -1,
+      'location': {'host': 'localhost', 'port': 8100, 'tlsPort': -1},
+      'dataSource': 'wikipedia_api',
+      'errorMsg': None
+    }
+  }
+  ```
+
+</details>
+
+### Get task segments
+
+#### URL
+
+<code class="getAPI">GET</code> `/druid/indexer/v1/task/{taskId}/segments`
 
 > This API is deprecated and will be removed in future releases.
 
-Retrieve information about the segments of a task.
+Retrieves information about segments generated by the task given the task id. To hit this endpoint, make sure to enable the audit log config on the Overlord with `druid.indexer.auditLog.enabled = true`.
+
+In addition to enabling audit log config, enable the cleanup of audit logs on the Coordinator. For reference, see [Audit records](../operations/clean-metadata-store.md#audit-records).
```

Comment:
This is a great addition. Are both of these required to access the endpoint? If so, I would phrase it more directly. For example:
```
To use this endpoint, you must enable the following:
* Enable the audit log...
* Enable the cleanup of audit logs...
```